### ISSUE TITLE
Duplicate Queries accessing HasOne relation that does not exist

### ISSUE THREAD

**Issue by jameshine** [laravel-filament/filament#5967]
### Package

filament/filament

### Package Version

v2.17.17

### Laravel Version

v.10.4.0

### PHP Version

PHP 8.2.2

### Problem description

I have a `User` and `Phone` model. A User HasOne Phone (following the Laravel example). When creating a hasOne relationship using a seeder, viewing the user record causes duplicate queries. If you try to create a User via the form, leaving both the fields below blank in this example, it'll throw the following:

`SQLSTATE[23000]: Integrity constraint violation: 19 NOT NULL constraint failed: phones.number (Connection: sqlite, SQL: insert into "phones" ("number", "serial_number", "user_id", "updated_at", "created_at") values (?, ?, 1, 2023-03-18 09:41:36, 2023-03-18 09:41:36))`

**Layout component with relationship**
```
Forms\Components\Section::make('phone')
                        ->relationship('phone')
                        ->schema([
                            Forms\Components\TextInput::make('number'),
                            Forms\Components\TextInput::make('serial_number'),
                        ])
```

https://user-images.githubusercontent.com/888834/226098158-884bd0e5-f8fa-459e-932b-19bd51af4b93.mov

<img width="1508" alt="Screenshot 2023-03-18 at 09 49 29" src="https://user-images.githubusercontent.com/888834/226098179-6dd97092-a5ea-40
<img width="1418" alt="Screenshot 2023-03-18 at 09 49 46" src="https://user-images.githubusercontent.com/888834/226098194-79efc701-0e5b-4b8d-9c48-bc246d240517.png">
04-8832-34cf10665386.png">

### Expected behavior

- One query to the phones table.
- The ability to save a user record without a phone since not all users have phones.

### Steps to reproduce

1. git clone https://github.com/jameshine/filament-has-one-example.git
2. cd filament-has-one-example
3. composer install
4. cp .env.example .env
5. touch database/database.sqlite
6. php artisan key:generate
7. php artisan migrate:fresh --seed
8. Login (test@example.com / password)
9. Visit admin/users/1/edit
10. Open debugbar to see duplicate queries.
11. Click save leaving both relation fields empty to see constraint violation.

### Reproduction repository

https://github.com/jameshine/filament-has-one-example.git

**Comment by danharrin** [comment]
The `not null` error is because the column in the DB isn't nullable. The fix for that one is to create a `reactive()` `Toggle` or `Checkbox` above the layout component which show/hides the layout component if the user doesn't want it (something like `visible(fn ($get) => $get('has_phone'))`). Then make the fields inside it required.

**Comment by jameshine** [comment]
> The `not null` error is because the column in the DB isn't nullable. The fix for that one is to create a `reactive()` `Toggle` or `Checkbox` above the layout component which show/hides the layout component if the user doesn't want it (something like `visible(fn ($get) => $get('has_phone'))`). Then make the fields inside it required.

Yes, that works, thank you. I’ve also used `saveRelationshipUsing` to update or delete the relation if that toggle is false.

This is now just the duplicate queries when no relation exists.

**Comment by danharrin** [comment]
Thank you very much for the excellent reproduction repository, it does help us a lot. Fixed by #6077.

**Comment by danharrin** [comment]
Apologies, but I've had to revert the fix. It caused #6107. I don't think there is a fix that allows relationships not to be saved twice, as we'd have to query the DB to be able to check if the record exists yet or not. Hence why I think this duplicate query is necessary. I don't think it should really cause any performance problem to be honest.
